File: backend-java/src/main/java/com/aiproject/common/BizException.java
package com.aiproject.common;

import lombok.Getter;

/**
 * 业务异常：携带业务错误码，由 GlobalExceptionHandler 统一转换为响应信封。
 */
@Getter
public class BizException extends RuntimeException {

    private final ErrorCode errorCode;

    public BizException(ErrorCode errorCode) {
        super(errorCode.getDefaultMessage());
        this.errorCode = errorCode;
    }

    public BizException(ErrorCode errorCode, String message) {
        super(message);
        this.errorCode = errorCode;
    }
}

File: backend-dotnet/AiProject.Api/Common/BizException.cs
namespace AiProject.Api.Common;

/// <summary>
/// 业务异常：携带错误码，由 ExceptionMiddleware 统一转换为响应信封。
/// </summary>
public class BizException : Exception
{
    public int Code { get; }

    public BizException(int code, string message) : base(message)
    {
        Code = code;
    }
}
